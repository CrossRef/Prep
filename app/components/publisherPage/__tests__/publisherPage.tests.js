import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';
import sinon from 'sinon'


import PublisherPage from '../publisherPage'



//Dummy Data
const participationSummary = {
  message: {
    "current-year": true,
    "totals": {
      "books": 1234,
        "conference-papers": 123,
        "reports": 123,
        "datasets": 0,
        "dissertations": 0,
        "posted-content": 0,
        "standards": 0
    },
    "Coverage": []
  }
}




//Mocks
const sandbox = sinon.sandbox.create()
global.fetch = sandbox.stub().callsFake(() => Promise.resolve({json: () => participationSummary}));
const didMount = sandbox.spy(PublisherPage.prototype, 'componentDidMount')
const setState = sandbox.spy(PublisherPage.prototype, 'setState')






//Render and begin tests
const mountComponent = mount(<PublisherPage match={{params: {publisherName: 'publisher name'}}}/>)


test('mount snapshot', () => {
  expect(didMount.callCount).toBe(1)
  expect(toJson(mountComponent)).toMatchSnapshot()
})


describe('componentDidMount', () => {
  test('should get totals from api', () => {
    expect(global.fetch.callCount).toBe(1)
    expect(setState.args[0][0]).toEqual({totals: participationSummary.message.totals})
    expect(mountComponent.state('totals')).toEqual(participationSummary.message.totals)
  })
})


//No way to test totalsTooltip hover functionality because it is CSS based