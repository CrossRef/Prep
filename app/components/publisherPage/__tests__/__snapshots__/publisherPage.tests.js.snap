// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount snapshot 1`] = `
<PublisherPage
  match={
    Object {
      "params": Object {
        "publisherName": "publisher name",
      },
    }
  }
>
  <div
    className="publisherPage"
  >
    <div
      className="topStripe"
    >
      <div
        className="widthContainer"
      >
        <div
          className="topBar"
        >
          <div
            className="button"
          >
            Find publisher
          </div>
          <div
            className="button"
          >
            Learn more
          </div>
        </div>
        <div
          className="contentBox"
        >
          <div
            className="leftBox"
          >
            <p
              className="firstText"
            >
              Comprehensive metadata makes publications discoverable.
            </p>
            <p
              className="blueText"
            >
              Make sure your content can be found.
            </p>
          </div>
          <div
            className="rightBox"
          >
            <div
              className="publisherTitle"
            >
              publisher name
            </div>
            <div
              className="totals"
            >
              <div
                className="totalTooltip"
              />
              <p
                className="totalNumber"
              >
                0
              </p>
              <p
                className="totalText"
              >
                Total registered content items
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="checksSection"
    >
      Checks go here
    </div>
    <div
      className="chartSection"
    >
      Chart goes here
    </div>
  </div>
</PublisherPage>
`;
